Author: Pino Toscano <pino@kde.org>
Description: Use GLIBC functions on any GLIBC architecture.
 Make use of __GLIBC__ instead of Q_OS_LINUX for functions available globally
 in GNU libc, not just on Linux.
Last-Update: 2010-10-02

--- a/src/corelib/tools/qlocale.cpp
+++ b/src/corelib/tools/qlocale.cpp
@@ -77,7 +77,7 @@ QT_END_NAMESPACE
 #include <qdebug.h>
 #include <time.h>
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 #    include <fenv.h>
 #endif
 
@@ -6637,7 +6637,7 @@ Q_CORE_EXPORT char *qdtoa ( double d, in
     _control87(MCW_EM, MCW_EM);
 #endif
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
     fenv_t envp;
     feholdexcept(&envp);
 #endif
@@ -6653,7 +6653,7 @@ Q_CORE_EXPORT char *qdtoa ( double d, in
 #endif //_M_X64
 #endif //Q_OS_WIN
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
     fesetenv(&envp);
 #endif
 
